cloudigrade/util/redhatcloud/sources.py

```diff
@@ -217,46 +224,40 @@ def notify_application_availability(
         )
         return
 
-    sources_api_base_url = settings.SOURCES_API_BASE_URL
-    sources_api_external_uri = settings.SOURCES_API_EXTERNAL_URI
-
-    url = (
-        f"{sources_api_base_url}/{sources_api_external_uri}"
-        f"applications/{application_id}/"
-    )
+    sources_kafka_config = {
+        "bootstrap.servers": f"{settings.LISTENER_SERVER}:{settings.LISTENER_PORT}"
+    }
 
     payload = {
-        "availability_status": availability_status,
-        "availability_status_error": availability_status_error,
+        "resource_type": settings.SOURCES_RESOURCE_TYPE,
+        "resource_id": application_id,
+        "status": availability_status,
+        "error": availability_status_error,
     }
 
     logger.info(
         _(
-            "Setting the availability status for application "
+            "Requesting the update of the availability status for application "
             "%(application_id)s as %(status)s"
         ),
         {"application_id": application_id, "status": availability_status},
     )
 
-    headers = identity.generate_http_identity_headers(account_number, is_org_admin=True)
-    response = requests.patch(url, headers=headers, data=json.dumps(payload))
-
-    logger.debug(
-        _("Notify sources response: %(code)s - %(response)s"),
-        {"code": response.status_code, "response": response.content},
-    )
+    try:
+        kafka_producer = KafkaProducer(sources_kafka_config)
 
-    if response.status_code == http.HTTPStatus.NOT_FOUND:
-        logger.info(
-            _(
-                "Cannot update availability status, application id "
-                "%(application_id)s not found."
-            ),
-            {"application_id": application_id},
+        kafka_producer.poll(0)
+        kafka_producer.produce(
+            topic=settings.SOURCES_STATUS_TOPIC,
+            value=json.dumps(payload),
+            headers={"event_type": settings.SOURCES_AVAILABILITY_EVENT_TYPE},
         )
-    elif response.status_code != http.HTTPStatus.NO_CONTENT:
-        message = _(
-            "Unexpected status {status} updating application "
-            "{application_id} status at {url}"
-        ).format(status=response.status_code, application_id=application_id, url=url)
-        raise SourcesAPINotOkStatus(message)
+        kafka_producer.flush()
+    except BufferError as error:
+        message = f"BufferError: {str(error)}"
+        logger.info(message)
```

Comment:
Are the `BufferError` and `KafkaException` exceptions expected to be raised in unusual error cases? If so, I'd be tempted to change this `logger.info` and the one below to `logger.exception`. That would result in a more immediate Sentry alert and would include the stack trace in the message that might help diagnose where things went wrong in the guts of `kafka_producer`'s calls.

Reply:
logger.exception it is 👍